src/plugins/intel_cpu/src/nodes/mkldnn_topk_node.cpp

```diff
@@ -1960,27 +1993,9 @@ void MKLDNNTopKNode::prepareParams() {
     }
 
     if (jit_mode) {
-        if (srcMemPtr->getDesc().hasLayoutType(LayoutType::ncsp)) {
-            layout = TopKLayoutType::topk_ncsp;
-        } else if (srcMemPtr->getDesc().hasLayoutType(LayoutType::nspc)) {
-            layout = TopKLayoutType::topk_nspc;
-        } else {
-            layout = TopKLayoutType::topk_blocked;
-        }
-
-        auto selectedPD = getSelectedPrimitiveDescriptor();
-        auto data_type = MKLDNNExtensionUtils::IEPrecisionToDataType(selectedPD->getConfig().inConfs[TOPK_DATA].desc->getPrecision());
-        data_size = MKLDNNExtensionUtils::sizeOfDataType(data_type);
-
-        topk_innermost = (layout == TopKLayoutType::topk_ncsp && axis == static_cast<int>(dst_dims.size() - 1)) ||
-                        ((layout == TopKLayoutType::topk_nspc || layout == TopKLayoutType::topk_blocked) && axis == 1);
-
-        if (mayiuse(cpu::x64::avx512_common)) {
-            blk_size = 16;
-            count_xmm = 16; // only 16 vector registers are valid in sse instructions
-        } else if (mayiuse(cpu::x64::sse41)) {
-            blk_size = 8;
-            count_xmm = 16;
+        if (!preset_params_done) {
+            preset_params();
```

Comment:
minor: Why we need to call preset_params here? I expect it will be always handled inside createPrimitive. In other words condition if `(!preset_params_done)` should be always false here. Or I missing smt?

Reply:
@dmitry-gorokhov Hi Dmitry, thanks for the comment! preset_params routine is called in prepareParams to set the variable "algorithm" and "bubble_inplace", which decide the  value of the flag "shape_agnostic_alg" used in subsequent prepare_original_idx routine. 
prepare_original_idx routine must be put inside prepareParams, because  heap sort and bubble sort already support shape agnostic (ticket 29913), they calculate original index tables inside prepareParams whenever shape changes. While Bitonic sort and inplaced bubble sort will still calculate these tables inside JIT kernel once for all, before they also support shape agnostic (ticket 76858). 
After ticket 76858 being applied, the flag shape_agnostic_alg will be useless, calculating original index tables should always been done inside prepareParams, no matter what algorithm is it. Then preset_params in prepareParams can be removed. Further discussion is always welcomed. Thanks Dmitry!